snapcraft/plugins/kbuild.py

```diff
@@ -112,16 +122,61 @@ def __init__(self, name, options, project):
         if logger.isEnabledFor(logging.DEBUG):
             self.make_cmd.append('V=1')
 
+    def unroll_ubuntu_config(self, config_path):
+        try:
+            with open(os.path.join('debian', 'debian.env'), 'r') as f:
+                env = f.read()
+        except OSError as e:
+            raise RuntimeError('Unable to access {}: {}'.format(e.filename,
+                                                                e.strerror))
+        arch = self.project.deb_arch
+        try:
+            branch = env.split('.')[1].strip()
+        except IndexError:
+            raise RuntimeError('Malformed debian.env, cannot extract'
+                               ' branch name')
+        flavour = self.options.kconfigflavour
+
+        configfiles = []
+        baseconfigdir = os.path.join('debian.{}'.format(branch), 'config')
+        archconfigdir = os.path.join('debian.{}'.format(branch),
+                                     'config', arch)
+        commonconfig = os.path.join(baseconfigdir,
+                                    'config.common.ports')
+        ubuntuconfig = os.path.join(baseconfigdir,
+                                    'config.common.ubuntu')
+        archconfig = os.path.join(archconfigdir,
+                                  'config.common.{}'.format(arch))
+        flavourconfig = os.path.join(archconfigdir,
+                                     'config.flavour.{}'.format(flavour))
+        configfiles.append(commonconfig)
+        configfiles.append(ubuntuconfig)
+        configfiles.append(archconfig)
+        configfiles.append(flavourconfig)
+        # assemble .config
+        try:
+            with open(config_path, 'w') as fw:
+                for f in configfiles:
+                    with open(f) as fr:
+                        fw.write(fr.read())
```

Comment:
fw, f and fr are bad variable names. At least following python conventions.
I would rewrite it something like this:

    try:
        with open(config_path, 'w') as config:
            for config_part_path in (commonconfig, ubuntuconfig, archconfig, flavourconfig):
                with open(config_part_path) as config_part:
                    config_file.write(config_part.read)